pkg/operator/controller/ingress/dns.go

```diff
@@ -29,11 +29,16 @@ const defaultRecordTTL int64 = 30
 
 // ensureWildcardDNSRecord will create DNS records for the given LB service.
 // If service is nil (haveLBS is false), nothing is done.
-func (r *reconciler) ensureWildcardDNSRecord(ic *operatorv1.IngressController, service *corev1.Service, haveLBS bool) (bool, *iov1.DNSRecord, error) {
+func (r *reconciler) ensureWildcardDNSRecord(ic *operatorv1.IngressController, platformStatus *configv1.PlatformStatus, dnsConfig *configv1.DNS, service *corev1.Service, haveLBS bool) (bool, *iov1.DNSRecord, error) {
 	if !haveLBS {
 		return false, nil, nil
 	}
 
+	if !manageDNSForDomain(ic.Status.Domain, platformStatus, dnsConfig) {
+		r.recorder.Eventf(ic, "Warning", "DomainNotMatching", fmt.Sprintf("Domain [%s] of ingresscontroller does not match the baseDomain [%s] of the cluster DNS config, so DNS management is not supported.", ic.Status.Domain, dnsConfig.Spec.BaseDomain))
```

Comment:
I'm ambivalent about emitting an event here.  On the one hand, it might be more visible and in some cases more persistent than status conditions or logs.  On the other hand, this will emit an event every time the ingresscontroller is reconciled, and the information is also conveyed in the ingresscontroller's status conditions.

Reply:
Is it possible to just emit an event when the status changes from DNS management supported to an unsupported state?

Reply:
That seems reasonable to me.  `syncIngressControllerStatus` could check some subset of status conditions (in this case, we just need to check the "DNSManaged" status condition) and emit an error if a status condition change value (in the case of "DNSManaged", this status condition should only change value, from uninitialized, the first time the ingresscontroller's status is updated after the ingresscontroller was created).

Reply:
Based on the discussion in this [thread](https://coreos.slack.com/archives/CCH60A77E/p1653662001871969), I moved the event to admission phase.